src/graph/executor/algo/BFSShortestPathExecutor.cpp

```diff
@@ -29,18 +29,26 @@ folly::Future<Status> BFSShortestPathExecutor::execute() {
   }
 
   std::vector<folly::Future<Status>> futures;
-  auto leftFuture = folly::via(runner(), [this]() { return buildPath(false); });
-  auto rightFuture = folly::via(runner(), [this]() { return buildPath(true); });
+  auto leftFuture = folly::via(runner(), [this]() {
+    memory::MemoryCheckGuard guard;
+    return buildPath(false);
+  });
+  auto rightFuture = folly::via(runner(), [this]() {
+    memory::MemoryCheckGuard guard;
+    return buildPath(true);
+  });
   futures.emplace_back(std::move(leftFuture));
   futures.emplace_back(std::move(rightFuture));
 
   return folly::collect(futures)
       .via(runner())
       .thenValue([this](auto&& status) {
+        memory::MemoryCheckGuard guard;
```

Comment:
I think you could enable it in `Schedueler::execute` and disable it in callback of `Future`.

Reply:
> I think you could enable it in `Schedueler::execute` and disable it in callback of `Future`.

`QueryInstance::execute` is the caller of `Schedueler::execute`, it already have MemoryCheckGuard on, see change in QueryInstance.cpp

The thread invoke thenValue's callback was another thread from folly, it should turn MemoryCheckGuard on by itself.